tests/ha/ha_cluster_init.pm

```diff
@@ -53,7 +53,11 @@ sub run {
     exec_csync;
 
     # State of SBD if shared storage SBD is used
-    assert_script_run "sbd -d $sbd_device list" unless (get_var('USE_DISKLESS_SBD'));
+    if (!get_var('USE_DISKLESS_SBD')) {
+        my $sbd_output = script_output("sbd -d $sbd_device list");
+        record_info('bsc#1170037', 'All nodes not shown by sbd list command', result => 'softfail')
+          if (get_node_number ne (my $clear_count = () = $sbd_output =~ /clear/g));
```

Comment:
This is beautiful. Only issue I can think of, and it's a minor one, is if one of the hostnames contains the word `clear`, then the `$clear_count` value will be different than the number of nodes. You could try with `/\sclear\s/` which should work as long as none of the nodes has a hostname identical to `clear`.

Another suggestion is to change `ne` for `==` as your comparing numbers, instead of strings.

Reply:
but for `ne` it should be `!=`, no?

Reply:
> but for `ne` it should be `!=`, no?

Yes, yes. It's that time of the day. I'll fix my comment

Reply:
Thanks!
You are right, I'm adding and testing your suggestions :smile: